toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -804,12 +794,16 @@ func FormatSinglePartition(partDevPath string, partition configuration.Partition
 	// To handle such cases, we can retry the command.
 	switch fsType {
 	case "fat32", "fat16", "vfat", "ext2", "ext3", "ext4", "xfs":
-		mkfsOptions := DefaultMkfsOptions[fsType]
-
 		if fsType == "fat32" || fsType == "fat16" {
 			fsType = "vfat"
 		}
 
+		mkfsOptions, err := getFileSystemOptions(targetOs, fsType)
+		if err != nil {
+			err = fmt.Errorf("failed to get mkfs args for filesystem type (%s):\n%w", fsType, err)
```

Comment:
nit - maybe it is worth logging the targetOS here too

Reply:
Fixed.